group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2653,7 +2641,7 @@ private boolean initializedAssignmentPending(ShareGroup group) {
         }
 
         // We need to check if all the group initialized share partitions are part of the group assignment.
-        Map<Uuid, Set<Integer>> initializedTps = shareGroupPartitionMetadata.get(group.groupId()).initializedTopics();
+        Map<Uuid, Set<Integer>> initializedTps = stripInitValue(shareGroupPartitionMetadata.get(group.groupId()).initializedTopics());
```

Comment:
nit: There is a mismatch in naming here. The record is `ShareGroupStatePartitionMetadataInfo` while the timeline hashmap is `shareGroupPartitionMetadata`. I wonder if `shareGroupStatePartitionMetadata` would be better.